pkg/ceph/osd/daemon.go

```diff
@@ -38,6 +38,11 @@ const (
 )
 
 func Run(context *clusterd.Context, agent *OsdAgent) error {
+	if err := setHostName(context); err != nil {
```

Comment:
should we be getting this from downstream API in kubelet? I'm not sure the environment is the best place to get a node identity.